src/App.tsx

```diff
@@ -21,19 +21,30 @@ function App() {
   const onSquareClickFunction = (i: number) => {
     console.log(i)
     let cloneBoardValues = [...boardValues];
-    if (cloneBoardValues[i] != '-') {
+    if (cloneBoardValues[i] !== '-') {
       return;
     }
     cloneBoardValues[i] = xIsNext ? 'O' : 'X';
     setBoardValues(cloneBoardValues);
     setXIsNext(!xIsNext);
 
-    console.log(calculateWinner(cloneBoardValues) + ' is the winner');
+    let winner = calculateWinner(cloneBoardValues);
+    console.log(winner);
+    const gameOver = cloneBoardValues.every((element: string) => element !== '-');
+    if (winner !== '-') {
+      displayWinner(winner);
+    } else if (gameOver && winner === null) {
+      displayWinner(winner);
+    }
+  }
```

Comment:
This code can be deleted--it's not actually doing anything.

Your `displayWinner` returns a string, and so the way you're using it in line 69 is correct-- rendering it in the React tsx code. 

Also, we already define const `winner` in line 14--- so we don't need to redefine it on line 31.